src/durableorchestrationcontext.ts

```diff
@@ -333,7 +370,7 @@ export class DurableOrchestrationContext {
      * @param fireAt The time at which the timer should expire.
      * @returns A TimerTask that completes when the durable timer expires.
      */
-    public createTimer(fireAt: Date): TimerTask {
+    public createTimer(fireAt: Date): TimerTask & TaskBase {

```

Comment:
The [`addNewChildren` method](https://github.com/Azure/azure-functions-durable-js/blob/01add5c8e01f1e3dcc7ad450852383ad933396ca/src/task.ts#L445) in `CallHttpWithPollingTask` expects a `TaskBase[]` argument, which is needed to use `child.parent`. This means `timerTask` returned by this call [here](https://github.com/Azure/azure-functions-durable-js/blob/01add5c8e01f1e3dcc7ad450852383ad933396ca/src/task.ts#L425) needs to be `TaskBase`-conforming, which is a fair assumption since both options returned by this method (`DFTimerTask` and `LongTimerTask` are both `TaskBase`

Reply:
> This looks great, but don't have a polling test yet, right?

Also, yes, still need to add unit tests but I verified with some manual testing and wanted to get some thoughts on the implementation and the questions I asked in the PR before adding the automated tests :)

Reply:
@hossam-nasr: Regarding the `& TaskBase` diff, I understand what you're saying but I still feel like there's a redundancy here. 

If `TimerTask` is a subtype of `TaskBase`, which do we need the `& TaskBase` assertion? Shouldn't the type-checker already be able to handle this?

Reply:
`TimerTask` is not a subtype of `TaskBase`, but just `Task`: https://github.com/Azure/azure-functions-durable-js/blob/a49c8ea2b6d4c7a3ee53c60d313cf752d6750b42/src/task.ts#L95, which has some missing properties from `TaskBase`. An alternative to this diff would be to make `TimerTask` extend `TaskBase` rather than `Task`. I have no preference as to which approach we take

Reply:
Although I'm not sure if an interface can extend an abstract class?

Reply:
Ah, I see now - I had forgotten that `TimerTask` did not inherit from `TaskBase`. 

Taking a step back: all the methods in the context method are public-facing, so I'd rather not modify their type signatures if possible.

With that in mind, I have 2 ideas.
1) You could refactor the implementation of `createTimer` to call a helper method, say `createTimerInternal`, which is only accessible in this scope (not to users) and that returns a `DFTimerTask` (which extends TaskBase while implementing TimerTask). Then `createTimer` can just return that value to users, and in your code _you_ could call that internal helper method to get the type you want.

2) You could just use a type assertion, i.e `as DFTimerTask` in your code to convince the type-checker that you can access the `parent` property of this Task.

Do these make sense? Let me know if these sound reasonable!

Reply:
Yes, that makes sense! I went with the second option since that required less changes :P 

For the record, the reason I wanted to call `createTimer` on the context rather than create a new `DFTimerTask` myself is on the off-chance that for some reason, actually a `LongTimerTask` is required here instead of a normal `DFTimerTask`. But both of them still conform to `TaskBase` so this shouldn't be an issue types-wise.